CloudFormation Modules (new  advanced module) (#166)
aws-samples/cfn101-workshop: content/advanced/modules/index.md

---
title: "Modules"
weight: 300
---

### Introduction

> This part of the workshop focuses on how you can extend creation, provisioning, and management capabilities of [AWS CloudFormation](https://aws.amazon.com/cloudformation/) with modules you develop.

Reviewer: ```suggestion
This part of the workshop focuses on how you can extend the creation, provisioning, and management capabilities of [AWS CloudFormation](https://aws.amazon.com/cloudformation/) with modules you develop.
```

> In this workshop so far you have seen how to use CloudFormation to build applications using the published resource types. In this lab you will leverage [AWS CloudFormation Modules](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/modules.html) to create reusable `modules` that you create once, and are used many times by other users in the same AWS Account and Region.

Reviewer: ```suggestion
In this workshop so far you have seen how to use CloudFormation to build applications using the resource types published by AWS. In this lab you will leverage [AWS CloudFormation Modules](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/modules.html) to create reusable template snippets that can be used many times by other users in the same AWS Account and Region.
```

A common use case for CloudFormation Modules is for codifying best practice or common configuration components that would be used within an enterprise, but might be company-specific, or contain proprietary logic.

A Module you create is treated as a first-class citizen within CloudFormation: you can manage your Module as you would manage any AWS resource. The Software Development Life Cycle (SDLC) process of a Module can be summarized as follows:

> 1. you install prerequisite tools you will use for development and testing of your Module;

Reviewer: "you" feels awkward in this list of steps. I think it would read better without it. "Install prerequisite tools for development..."

Author: Agreed, i'll make the change. I was a "victim" of using the same block of text from the Resource Types module ;)
